dwave/cloud/exceptions.py

```diff
@@ -22,23 +22,48 @@ class ConfigFileParseError(ConfigFileError):
     """Invalid format of config file."""
 
 
-class SolverError(Exception):
+class SAPIError(Exception):
+    """Generic SAPI error base class."""
+
+    def __init__(self, *args, **kwargs):
+        self.error_msg = kwargs.pop('error_msg', None)
+        self.error_code = kwargs.pop('error_code', None)
+        super(SAPIError, self).__init__(*args, **kwargs)
+
+    def __str__(self):
+        return super(SAPIError, self).__str__() or self.error_msg or ''
+
+class ResourceAuthenticationError(SAPIError):
+    """Access to resource not authorized."""
+
+class ResourceNotFoundError(SAPIError):
+    """Resource does not exist."""
```

Comment:
Just wondering if this is accurate. Not finding a resource might not mean it doesn't exist.

Reply:
If the client goes out to find it, and it can't, it surely does not exist. It can't. It must not. :scream: